sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/text/TextSuite.scala

```diff
@@ -172,6 +172,14 @@ class TextSuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("SPARK-23148: test for spaces in file names") {
```

Comment:
Shall we remove this test case and only leave test CSV / JSON with `multiLine` enabled? I think other tests are basically duplicates of https://github.com/apache/spark/blob/a0aedb0ded4183cc33b27e369df1cbf862779e26/sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala#L71.

Reply:
@HyukjinKwon good idea, thanks for pointing out that test - how about I just add a space to the special characters string, and have the existing test also test multiline on/off for text, csv and json?

Reply:
Let's just add both `multiLine` specific tests for `CSVSuite` and `JsonSuite` and add a space in "SPARK-22146 read files containing special characters using ..." if you prefer this way.

Reply:
Alternatively, adding the test like below in `FileBasedDataSourceSuite.scala`, instead of `CSVSuite` and `JsonSuite`:

```scala
// Only CSV/JSON supports multiLine option and the code paths are different blabla ..
Seq("csv", "json").foreach { format =>
  test(s"SPARK-23148 read files containing special characters using $format - multLine enabled") {
    val nameWithSpecialChars = s"sp&ci al%chars"
    withTempDir { dir =>
      ... spark.read.option("multuLine", true).format(format)...
    }
  }
}
```

could be also fine if possible. Either way is fine to me.

Reply:
In the end, to reduce code duplication, I made it so that orc and parquet run multiline as well (I tried to find a neat way to only run multiline if the format was csv, text or json without having a separate test case but it just complicated things). Let me know if you'd rather I have two separate test cases to avoid running the two redundant cases with orc / parquet.